tests/boot/boot_from_pxe.pm

```diff
@@ -155,23 +155,27 @@ sub run {
         my $ssh_vnc_wait_time = 300;
         my $ssh_vnc_tag       = eval { check_var('VIDEOMODE', 'text') ? 'sshd' : 'vnc' } . '-server-started';
         my @tags              = ($ssh_vnc_tag, 'orthos-grub-boot-linux');
-        assert_screen \@tags, $ssh_vnc_wait_time;
-        save_screenshot;
-        sleep 2;
 
-        if (match_has_tag("orthos-grub-boot-linux")) {
-            my $image_name = eval { check_var("INSTALL_TO_OTHERS", 1) ? get_var("REPO_0_TO_INSTALL") : get_var("REPO_0") };
-            my $args       = "initrd auto/openqa/repo/${image_name}/boot/${arch}/initrd";
-            wait_still_screen 5;
-            type_string $args;
-            send_key 'ret';
-            assert_screen 'orthos-grub-boot-initrd', $ssh_vnc_wait_time;
-            $args = "boot";
-            type_string $args;
-            send_key "ret";
-            assert_screen $ssh_vnc_tag, $ssh_vnc_wait_time;
+        # Proceed if the 'installation' console is ready
+        # otherwise the 'sol' console may be just freezed
+        if (check_screen(\@tags, $ssh_vnc_wait_time)) {
```

Comment:
I am afraid , to let the original logic work, it should be assert_screen, or for ssh installation which needs to wait needle sshd-server-started show up , then select installation, will meet problem sometimes.

Reply:
Ok, get the point. So you want a try to reach installation console anyway to gain more success chance.